lib/puppet/functions/unwrap.rb

```diff
@@ -29,12 +30,12 @@
 #
 Puppet::Functions.create_function(:unwrap) do
   dispatch :unwrap do
-    param 'Sensitive', :arg
+    param 'Any', :arg
```

Comment:
Perhaps it would be cleaner to have 2 dispatch methods? So it would become
```ruby
dispatch :unwrap_sensitive do
  param 'Sensitive', :arg
  optional_block_param
end

dispatch :unwrap_other do
  param 'Any', :arg
  optional_block_param
end
```

Reply:
> Perhaps it would be cleaner to have 2 dispatch methods?   

Suggestion implemented.

Reply:
I just wonder, if `dispatch` is deterministic here by only the Order of the two Declarations.  Because `Any` would also match `Sensitive`…

Reply:
Yes, that's something I assumed but https://puppet.com/docs/puppet/7/functions_ruby_signatures.html isn't very explicit in it.

Reply:
According to [4.x function specification](https://github.com/puppetlabs/puppet-specifications/blob/8db1ac65aa6cc6d84bbe0c366ee4b9378d9bfcbd/language/func-api.md#defining-a-typed-dispatch) the behavior is sensitive to the order:

> the signatures are tested in the order they appear in the definition of the function, thus, if a very generic entry is placed first it will always win.

Reply:
👍

Reply:
I don't have permission to mark this as resolved, but this is implemented now.